source/common/api/win32/os_sys_calls_impl.cc

```diff
@@ -114,8 +114,14 @@ SysCallIntResult OsSysCallsImpl::ioctl(os_fd_t sockfd, unsigned long control_cod
 }
 
 SysCallIntResult OsSysCallsImpl::close(os_fd_t fd) {
-  const int rc = ::closesocket(fd);
-  return {rc, rc != -1 ? 0 : ::WSAGetLastError()};
+  int rc = ::closesocket(fd);
+  const int socket_err = ::WSAGetLastError();
```

Comment:
this doesn't appear to be called out in the release note.  is it tested in the new test?

Reply:
Discussed in Slack; in existing code close is only called where closesocket behavior is fine. My hope is to later change this back and make it *explicitly* closesocket (for all platforms), and remove everything I've done here after implementing [fstat, pread, pwrite, tmpfiles] in `Envoy::Filesystem::File` and migrating `AsyncFileManager` to using that. But getting it in here allows me to make progress elsewhere while that chain of changes is in flight.